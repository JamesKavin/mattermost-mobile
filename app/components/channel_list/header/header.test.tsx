// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {SafeAreaProvider} from 'react-native-safe-area-context';

import {renderWithIntl} from '@test/intl-test-helper';

import Header from './header';

describe('components/channel_list/header', () => {
    it('Channel List Header Component should match snapshot', () => {
        const {toJSON} = renderWithIntl(
            <SafeAreaProvider>
                <Header
                    canCreateChannels={true}
                    canJoinChannels={true}
                    displayName={'Test!'}
                />
            </SafeAreaProvider>,
        );

        expect(toJSON()).toMatchSnapshot();
    });
});
