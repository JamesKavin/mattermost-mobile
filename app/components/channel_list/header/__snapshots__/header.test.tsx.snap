// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/channel_list/header Channel List Header Component should match snapshot 1`] = `
<View
  animatedStyle={
    Object {
      "value": Object {
        "marginLeft": 0,
      },
    }
  }
  collapsable={false}
  style={
    Object {
      "marginLeft": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "Metropolis-SemiBold",
              "fontSize": 28,
              "fontWeight": "600",
              "lineHeight": 36,
            }
          }
          testID="channel_list_header.team_display_name"
        >
          Test!
        </Text>
        <View
          style={
            Object {
              "marginLeft": 4,
            }
          }
          testID="channel_list_header.chevron.button"
        >
          <Icon
            name="chevron-down"
            style={
              Object {
                "color": "rgba(255,255,255,0.8)",
                "fontSize": 24,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255,255,255,0.08)",
          "borderRadius": 14,
          "height": 28,
          "justifyContent": "center",
          "opacity": 1,
          "width": 28,
        }
      }
      testID="channel_list_header.plus.button"
    >
      <Icon
        name="plus"
        style={
          Object {
            "color": "rgba(255,255,255,0.8)",
            "fontSize": 18,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "rgba(255,255,255,0.64)",
        "fontFamily": "OpenSans-SemiBold",
        "fontSize": 11,
        "fontWeight": "600",
        "lineHeight": 16,
      }
    }
    testID="channel_list_header.server_display_name"
  >
    
  </Text>
</View>
`;
